scipy/optimize/optimize.py

```diff
@@ -3061,6 +2912,576 @@ def show_options(solver=None, method=None, disp=True):
+        self._hyper = {}
+        self.nit = 0
+        self.maxcv = np.nan
+
+    @property
+    def hyper_parameters(self):
+        """Dictionary containing solver hyperparameters."""
+        return self._hyper
+
+    @property
+    def nfev(self):
+        return self.func.nfev
+
+    @property
+    def njev(self):
+        return self.func.njev
+
+    @property
+    def nhev(self):
+        return self.func.nhev
+
+    @property
+    def result(self):
+        """
+        The optimization result represented as a `OptimizeResult` object.
+        Important attributes are: `x` the solution array, `success` a Boolean
+        flag indicating if the optimizer finished successfully and `message`
+        which describes the cause of the termination. See `OptimizeResult`
+        for a description of other attributes.
+        """
+        # override to set other fields.
+        if self.nit == 0:
+            return None
+
+        result = OptimizeResult(
+            x=self.x,
+            fun=self.fun,
+            nfev=self.nfev,
+            nit=self.nit,
+            message=self.message,
+            success=(self.warning_flag == 0 and self.converged()))
+        return result
+
+    def __call__(self, iterations):
+        """
+        Advance the solver by a number of steps.
+
+        Parameters
+        ----------
+        iterations: int
+            The number of iterations to perform
+
+        Returns
+        -------
+        result: OptimizeResult
+            The optimization result represented as a OptimizeResult object.
+        """
+        # TODO interupt once convergence is reached?
+        for it in range(iterations):
+            self.x, self.fun = next(self)
+
+        self._finish_up()
+        return self.result
+
+    @property
+    def N(self):
+        """
+        The dimensionality of the problem, `np.size(x)`.
+        """
+        if self.x is not None:
+            return self.x.size
+        elif self.x0 is not None:
+            return self.x0.size
+        return RuntimeError("Cannot determine problem size at this time,"
+                            " perform at least one iteration")
+
+    def solve(self):
+        """
+        Run the solver through to completion.
+
+        Returns
+        -------
+        result: OptimizeResult
+            The optimization result represented as a `OptimizeResult` object.
+            Important attributes are: `x` the solution array, `success` a
+            Boolean flag indicating if the optimizer exited successfully and
+            `message` which describes the cause of the termination. See
+            `OptimizeResult` for a description of other attributes.
+        """
+        for x, fun in self:
+            self.x, self.fun = x, fun
+
+            if (self.converged() or self.nfev >= self.maxfun or
+                    self.nit >= self.maxiter):
+                break
+
+        self._finish_up()
+        return self.result
+
+    def converged(self):
```

Comment:
Maybe we should implement this. We pass `maxiter` and  `maxfun` to the Optimizer; should we use them?

Reply:
Hmm. Personally I don't think `self.nit >=self.maxiter` belong in `converged`. For example, you may set up a maxiter that's too low to reach convergence, but then still want to do more iterations.

Reply:
> Personally I don't think `self.nit >=self.maxiter` belong in `converged`

Fair point. Then I think it's best we leave it unimplemented. We could define another function, but I'd rather not have the overhead.

Reply:
After some changes the only place where maxiter and maxfun reside are in the solve method.

Reply:
I think that's fair. I'd rather provide a framework and let the user decide.